### ISSUE TITLE
Fails on very basic scenario at v0.9.0

### ISSUE THREAD

**Issue by DmitryBatalov** [fsprojects/SwaggerProvider#94]
### Description

Fails on very basic scenario. No issue in 0.8.1.

### Repro steps

1. Create console application.
2. Add  reference to SwaggerProvider 0.9.0
3. Add script.fsx with following code (don't forget to add it to project file)
```
#load "../packages/SwaggerProvider/SwaggerProvider.fsx" 

open SwaggerProvider

type PetStore = SwaggerProvider<"http://petstore.swagger.io/v2/swagger.json">
```
4. Send it to  REPL and you'll get an error.

### Expected behavior

No issue 

### Actual behavior
REPL reports 
```
script.fsx(6,17): error FS3033: The type provider 'SwaggerProvider.SwaggerTypeProvider' reported an error: The design-time type 'FSharp.Data.Http' utilized by a type provider was not found in the target reference assembly set '[|tgt assembly Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed;
  tgt assembly YamlDotNet, Version=4.2.2.0, Culture=neutral;
  tgt assembly SwaggerProvider.Runtime, Version=0.9.0.0, Culture=neutral;
  tgt assembly SwaggerProvider, Version=0.9.0.0, Culture=neutral;
  tgt assembly mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;
  tgt assembly FSharp.Core, Version=4.4.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;
  tgt assembly System.Xml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;
  tgt assembly System.Runtime.Remoting, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;
  tgt assembly System.Runtime.Serialization.Formatters.Soap, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Data, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;
  tgt assembly System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;
  tgt assembly System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Linq, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Reflection, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Linq.Expressions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Threading.Tasks, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.IO, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Net.Requests, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Collections, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Runtime.Numerics, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Threading, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.ValueTuple, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51;
  tgt assembly System.Web, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Web.Services, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tgt assembly System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;
  tgt assembly System.Numerics, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;
  tgt assembly FSharp.Compiler.Interactive.Settings, Version=4.4.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;
  tmp46F7|]'. You may be referencing a profile which contains fewer types than those needed by the type provider you are using.
```

### Known workarounds

Downgrade to 0.8.1 doesn't have such issue.

### Related information 

* Operating system Win10
* .NET Runtime

**Comment by sergey-tihon** [comment]
This particular issue should be fixed in [v0.10.0-alpha3](https://www.nuget.org/packages/SwaggerProvider/0.10.0-alpha3)

But now we have another very interesting issue with TP in FSI

> C:\Users\sergey\AppData\Local\Temp\unknown(1,1): error FS3216: type 'SwaggerProvider.VismaApi+PaginatedResponse[AccountApi]' not found in assembly 'tmp760, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. A possible cause may be a version incompatibility. You may need to explicitly reference the correct version of this assembly to allow all referenced components to use the correct version.

Repro script
```fsharp
#load @"packages/SwaggerProvider/SwaggerProvider.fsx"
open SwaggerProvider

[<Literal>]
let schema = @"https://eaccountingapi-sandbox.test.vismaonline.com/swagger/docs/v2"

type VismaApi = SwaggerProvider<schema>
let api = VismaApi()

// NOTE: Error appears during evaluation of this line
let customers = api.AccountsV2Get()
```

but at the same time, it works if compiled into the console application
![2017-12-01_1953](https://user-images.githubusercontent.com/1197905/33493561-7d72e174-d6d1-11e7-8c43-37bcbc2fd919.png)

**Comment by sergey-tihon** [comment]
`v0.9.0` is unlisted because it was really broken 😞 

Full fix is provided in [0.10.0-alpha4](https://www.nuget.org/packages/SwaggerProvider/0.10.0-alpha4)

FSI cannot load type with square brackets in name like `PaginatedResponse[AccountApi]` so you @DmitryBatalov a unique schema =) 

`v0.10.0-alpha4` generates nicer names now - `PaginatedResponseAccountApi`

![2017-12-01_2309](https://user-images.githubusercontent.com/1197905/33501289-1d024cbe-d6ed-11e7-9600-0b5e75b45999.png)
